[Issue] panic when received SIGINT

### Current Behavior

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x118 pc=0x7ccdd2]

goroutine 1 [running]:
github.com/regclient/regclient/types/blob.(*BReader).Close(0xc0003b0df0?)
        github.com/regclient/regclient/types/blob/reader.go:85 +0x12
github.com/regclient/regclient/mod.Apply.func3.1()
        github.com/regclient/regclient/mod/mod.go:121 +0x24
github.com/regclient/regclient/mod.Apply.func3(0xc0004c6c00)
        github.com/regclient/regclient/mod/mod.go:136 +0x374
github.com/regclient/regclient/mod.dagWalkLayers(0xc00037ac60, 0xc0003b1408)
        github.com/regclient/regclient/mod/dag.go:509 +0x92
github.com/regclient/regclient/mod.dagWalkLayers(0xc000446420, 0xc0003b1408)
        github.com/regclient/regclient/mod/dag.go:498 +0x137
github.com/regclient/regclient/mod.Apply({0xa9a778, 0xc0000b04b0}, _, {{0x9a957b, 0x3}, {0x7ffcff334c3d, 0x15}, {0x9aaf58, 0x9}, {0x7ffcff334c3d, ...}, ...}, ...)
        github.com/regclient/regclient/mod/mod.go:117 +0x750
main.(*imageCmd).runImageMod(0xc00033a160, 0xc000360608, {0xc00031e150?, 0x0?, 0x0?})
        github.com/regclient/regclient/cmd/regctl/image.go:1651 +0x77f
github.com/spf13/cobra.(*Command).execute(0xc000360608, {0xc00031e0e0, 0x7, 0x7})
        github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc000320008)
        github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        github.com/spf13/cobra@v1.8.0/command.go:1032
main.main()
        github.com/regclient/regclient/cmd/regctl/main.go:29 +0x14e
```

### Expected Behavior

Won't panic

### Steps To Reproduce

1. `regctl -v error image mod fluent/fluent-bit:1.8 --layer-compress none --create ocidir://fluent-bit:1.8`
2. Press control+c

### Version

```text
VCSTag:     v0.6.1
VCSRef:     766ee6291f882778207ff42207f9ca8b1da54e57
VCSCommit:  766ee6291f882778207ff42207f9ca8b1da54e57
VCSState:   clean
VCSDate:    2024-05-14T13:18:19Z
Platform:   linux/amd64
GoVer:      go1.22.3
GoCompiler: gc
```

### Environment

- Running as binary or container: binary
- Host platform: linux/amd64
- Registry description: dockerhub

### Anything else

nope

--- a/mod/mod.go
+++ b/mod/mod.go
@@ -131,10 +131,11 @@ func Apply(ctx context.Context, rc *regclient.RegClient, rSrc ref.Ref, opts ...O
 				return dl, nil
 			}
 			if len(dc.stepsLayer) > 0 {
-				rdr, err = rc.BlobGet(ctx, rSrc, dl.desc)
+				bRdr, err := rc.BlobGet(ctx, rSrc, dl.desc)
 				if err != nil {
 					return nil, err
 				}
+				rdr = bRdr
 				for _, sl := range dc.stepsLayer {
 					rdrNext, err := sl(ctx, rc, rSrc, rTgt, dl, rdr)
 					if err != nil {
@@ -148,10 +149,11 @@ func Apply(ctx context.Context, rc *regclient.RegClient, rSrc ref.Ref, opts ...O
 					return dl, nil
 				}
 				if rdr == nil {
-					rdr, err = rc.BlobGet(ctx, rSrc, dl.desc)
+					bRdr, err := rc.BlobGet(ctx, rSrc, dl.desc)
 					if err != nil {
 						return nil, err
 					}
+					rdr = bRdr
 				}
 				changed := false
 				empty := true

--- a/types/blob/blob_test.go
+++ b/types/blob/blob_test.go
@@ -227,6 +227,29 @@ func TestCommon(t *testing.T) {
 }
 
 func TestReader(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var b *BReader
+		_, err := b.Read([]byte{})
+		if err == nil {
+			t.Errorf("nil read did not fail")
+		}
+		err = b.Close()
+		if err != nil {
+			t.Errorf("nil close failed: %v", err)
+		}
+		_, err = b.ToOCIConfig()
+		if err == nil {
+			t.Errorf("nil convert ot OCI did not fail")
+		}
+		_, err = b.ToTarReader()
+		if err == nil {
+			t.Errorf("nil convert to tar reader did not fail")
+		}
+		_, err = b.Seek(0, io.SeekStart)
+		if err == nil {
+			t.Errorf("nil seek did not fail")
+		}
+	})
 	t.Run("empty", func(t *testing.T) {
 		// create empty blob
 		b := NewReader()

--- a/types/blob/reader.go
+++ b/types/blob/reader.go
@@ -82,7 +82,7 @@ func NewReader(opts ...Opts) *BReader {
 
 // Close attempts to close the reader and populates/validates the digest.
 func (r *BReader) Close() error {
-	if r.origRdr == nil {
+	if r == nil || r.origRdr == nil {
 		return nil
 	}
 	// attempt to close if available in original reader
@@ -100,7 +100,7 @@ func (r *BReader) RawBody() ([]byte, error) {
 
 // Read passes through the read operation while computing the digest and tracking the size.
 func (r *BReader) Read(p []byte) (int, error) {
-	if r.reader == nil {
+	if r == nil || r.reader == nil {
 		return 0, fmt.Errorf("blob has no reader: %w", io.ErrUnexpectedEOF)
 	}
 	size, err := r.reader.Read(p)
@@ -133,6 +133,9 @@ func (r *BReader) Seek(offset int64, whence int) (int64, error) {
 	if offset != 0 || whence != io.SeekStart {
 		return r.readBytes, fmt.Errorf("unable to seek to arbitrary position")
 	}
+	if r == nil || r.origRdr == nil {
+		return 0, fmt.Errorf("blob has no reader")
+	}
 	rdrSeek, ok := r.origRdr.(io.Seeker)
 	if !ok {
 		return r.readBytes, fmt.Errorf("Seek unsupported")
@@ -159,7 +162,7 @@ func (r *BReader) Seek(offset int64, whence int) (int64, error) {
 
 // ToOCIConfig converts a BReader to a BOCIConfig.
 func (r *BReader) ToOCIConfig() (*BOCIConfig, error) {
-	if !r.blobSet {
+	if r == nil || !r.blobSet {
 		return nil, fmt.Errorf("blob is not defined")
 	}
 	if r.readBytes != 0 {
@@ -184,7 +187,7 @@ func (r *BReader) ToOCIConfig() (*BOCIConfig, error) {
 
 // ToTarReader converts a BReader to a BTarReader
 func (r *BReader) ToTarReader() (*BTarReader, error) {
-	if !r.blobSet {
+	if r == nil || !r.blobSet {
 		return nil, fmt.Errorf("blob is not defined")
 	}
 	if r.readBytes != 0 {

